### PR TITLE
 Merge BottomCornerText

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: src/st/wrp/popup.c [hunk 1]
```diff
@@ -138,3 +139,186 @@ void BottomCornerText(u8* str, u8 lower_left) {
...
+
+    g_unkGraphicsStruct.BottomCornerTextPrims =
+        g_api.AllocPrimitives(PRIM_SPRT, charcount + 4);
+    if (g_unkGraphicsStruct.BottomCornerTextPrims == -1) {
+        return;
+    }
+#undef charcount
+
+    prim = &g_PrimBuf[g_unkGraphicsStruct.BottomCornerTextPrims];
+    prim->type = 3;
```
Comment (Xeeynamo) [inline src/st/wrp/popup.c:214]:
```suggestion
    prim->type = PRIM_G4;
```

File: src/st/wrp/popup.c [hunk 2]
```diff
@@ -138,3 +139,186 @@ void BottomCornerText(u8* str, u8 lower_left) {
...
+        if (ch == 0xFF) {
+            ch = *str++;
+
+            if (ch == 0) {
+                break;
+            }
+        } else {
+
+            (void)1;
+        }
```
Comment (Xeeynamo) [inline src/st/wrp/popup.c:175]: I noticed from your scratch you can remove this whole `else` block

File: src/st/wrp/popup.c [hunk 3]
```diff
@@ -138,3 +139,186 @@ void BottomCornerText(u8* str, u8 lower_left) {
...
+    prim->y0 = 0xCB;
+    prim->u0 = 0xA8;
+    prim->v0 = 0;
+    prim->u1 = 0x10;
+    prim->v1 = 0x18;
+    prim->priority = 0x1EF;
+    prim->drawMode = DRAW_DEFAULT;
+    prim = prim->next;
+
+    prim->type = 4;
```
Comment (Xeeynamo) [inline src/st/wrp/popup.c:268]:
```suggestion
    prim->type = PRIM_GT4;
```

File: src/st/wrp/popup.c [hunk 4]
```diff
@@ -138,3 +139,186 @@ void BottomCornerText(u8* str, u8 lower_left) {
...
+            prim->u0 = (s8)((ch & 0xF) * 8);
+            prim->v0 = (s8)((ch & 0xF0) >> 1);
+            prim->u1 = prim->v1 = 8;
+            prim->priority = 0x1F0;
+            prim->drawMode = DRAW_DEFAULT;
+            prim->x0 = xpos2;
+            prim->y0 = 0xcc;
+            prim = prim->next;
+        }
+        if (((ch) >= 0xF0) && ((ch) < 0xFD)) {
```
Comment (Xeeynamo) [inline src/st/wrp/popup.c:315]: Can we remove the parentheses to make that `(ch)` to `ch`? I personally feel they add a bit too much unnecessary noise

**Review by Xeeynamo — APPROVED** [review]
Super! Left some minor comments

**Comment by sozud** [comment]
Comments should be resolved now

**Pull request merged**